### DISCUSSION TITLE
The given address is not yours

### DISCUSSION THREAD

**Discussion by Nalem14** [QuorumDMS/ftp-srv#217] [Q&A]
Hello, when I try to login to my FTP server, I have this error.

**FileZila Logs:**
`Statut :         	Résolution de l'adresse de thor.orion-serv.fr
Statut :         	Connexion à 37.187.141.180:2122...
Statut :         	Connexion établie, attente du message d'accueil...
Statut :         	Serveur non sécurisé, celui-ci ne supporte pas FTP sur TLS.
Statut :         	Connecté
Statut :         	Récupération du contenu du dossier...
Commande :	PWD
Réponse :     	257 "/"
Commande :	TYPE I
Réponse :     	200 Switch to "binary" transfer mode.
Commande :	PASV
Réponse :     	227 PASV OK (ftp://0,0,0,0,70,80)
Commande :	PORT 192,168,1,91,253,224
Réponse :     	500 The given address is not yours
Erreur :         	Impossible de récupérer le contenu du dossier
Statut :         	Connexion interrompue par le serveur`

**FTP-SRV Logs:**
`{"name":"ftp-srv","hostname":"thor.orion-serv.fr","pid":22617,"id":"50299f64-4a15-4a07-8a17-458b421d1f76","directive":"PORT","level":50,"err":{"message":"The given address is not yours","name":"SocketError","stack":"SocketError: The given address is not yours\n    at closeExistingServer.then (/home/orion/public_html/www3/server/node_modules/ftp-srv/src/connector/active.js:33:15)\n    at tryCatcher (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/promise.js:547:31)\n    at Promise._settlePromise (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/promise.js:604:18)\n    at Promise._settlePromiseCtx (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/promise.js:641:10)\n    at _drainQueueStep (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/async.js:97:12)\n    at _drainQueue (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/async.js:86:9)\n    at Async._drainQueues (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/async.js:102:5)\n    at Immediate.Async.drainQueues [as _onImmediate] (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/async.js:15:14)\n    at runCallback (timers.js:705:18)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)","code":500},"msg":"The given address is not yours","time":"2020-09-07T12:20:51.217Z","v":0}`

**Configs**
`const ftpServer = new FtpSrv({ 
	url: "ftp://0.0.0.0:2122",
	pasv_url: 'ftp://0.0.0.0',
	pasv_min: 18000,
	pasv_max: 18999,
	greeting: "Bienvenue sur le serveur FTP",
	file_format: "ls",
	timeout: 30000
});`

**Comment by Nalem14** [comment]
Resolved by setting "Auto detect external IP from FileZilla website" on FileZilla.

Now, another error.
```
Statut :         	Résolution de l'adresse de thor.orion-serv.fr
Statut :         	Connexion à 37.187.141.180:2122...
Statut :         	Connexion établie, attente du message d'accueil...
Statut :         	Serveur non sécurisé, celui-ci ne supporte pas FTP sur TLS.
Statut :         	Connecté
Statut :         	Récupération du contenu du dossier...
Commande :	PWD
Réponse :     	257 "/"
Commande :	TYPE I
Réponse :     	200 Switch to "binary" transfer mode.
Commande :	PASV
Réponse :     	227 PASV OK (ftp://0,0,0,0,4,1)
Commande :	PORT 77,202,201,191,255,151
Réponse :     	200 The requested action has been successfully completed
Commande :	LIST
Réponse :     	425 No connection established
Erreur :         	Impossible de récupérer le contenu du dossier
Statut :         	Connexion interrompue par le serveur
```

```
{"name":"ftp-srv","hostname":"thor.orion-serv.fr","pid":8310,"id":"7a1dea83-4320-48cb-be5f-40d17fbaa29e","directive":"LIST","level":50,"err":{"message":"operation timed out","name":"TimeoutError","stack":"TimeoutError: operation timed out\n    at afterTimeout (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/timers.js:46:19)\n    at Timeout.timeoutTimeout [as _onTimeout] (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/timers.js:76:13)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at listOnTimeout (timers.js:263:5)\n    at Timer.processTimers (timers.js:223:10)"},"msg":"operation timed out","time":"2020-09-07T12:43:36.402Z","v":0}
data:    /home/orion/public_html/www3/server/panel.js:8310 - {"name":"ftp-srv","hostname":"thor.orion-serv.fr","pid":8310,"id":"f07401a8-7dc4-446c-b7d3-cc9d2e013074","directive":"LIST","level":50,"err":{"message":"operation timed out","name":"TimeoutError","stack":"TimeoutError: operation timed out\n    at afterTimeout (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/timers.js:46:19)\n    at Timeout.timeoutTimeout [as _onTimeout] (/home/orion/public_html/www3/server/node_modules/ftp-srv/node_modules/bluebird/js/release/timers.js:76:13)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at listOnTimeout (timers.js:263:5)\n    at Timer.processTimers (timers.js:223:10)"},"msg":"operation timed out","time":"2020-09-07T12:44:35.292Z","v":0}
```

**Comment by trs** [comment]
The `pasv_url` option is actually a hostname, not a URI like `url` option (bad name, I know).
This `pasv_url` must also be the hostname of the server to give the client so it can make a connection to. You can see in the second set of logs there is a `PASV` request followed by a `PORT` request. I'm assuming this is because Filezilla is unable to connect to the invalid host provided by the server's `PASV` response.
The `PORT` command instructs the server to connect to the client instead, which is where the first log's `The given address is not yours` comes into play: these connections must be to the same endpoint. See https://github.com/autovance/ftp-srv/security/advisories/GHSA-jw37-5gqr-cf9j

Maybe the server gets into a weird state by having a `PASV` followed by a `PORT` command? Although it should correctly create a new data connection. Can you try forcing Filezilla to make a direct active connection, avoiding the `PASV` all together, and see if that works?
